internal/controllers/topology/cluster/current_state.go

```diff
@@ -119,6 +120,16 @@ func (r *Reconciler) getCurrentControlPlaneState(ctx context.Context, cluster *c
 		return nil, errors.Wrapf(err, "failed to get InfrastructureMachineTemplate for %s", tlog.KObj{Obj: res.Object})
 	}
 
+	mhc := &clusterv1.MachineHealthCheck{}
+	// MachineHealthCheck always has the same name and namespace as the ControlPlane object it belongs to.
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: res.Object.GetNamespace(), Name: res.Object.GetName()}, mhc); err != nil {
```

Comment:
Yeah - I think the current state can diverge from the blueprint here which may be necessary during deletion. It is a lot of get calls though (once every reconcile for something that 'probably' doesn't exist.) With cache etc. I don't think the lookup is an issue, but it bears thinking about.

Reply:
Yup agree. All things considered I would currently prefer more get calls to the cache vs. potentially leaking MHC objects which should have been deleted, but the deletion failed in some weird way.

Reply:
Just realized that it wouldn't even work. Let's assume we remove an MHC from the ClusterClass then we have to still get the MHC on the reconcile of that removal. If we would check here for the MHC it would be already removed from the control plane configuration...

Reply:
So I think let's resolve :)

Reply:
Why does github only ever hide the unresolved issues 😠

Reply:
We should read it always because it could be that now it is not configured in the ClusterClass, but before it was (and so we have to delete)